code/game/gamemodes/changeling/implements/hivemind.dm

```diff
@@ -0,0 +1,64 @@
+/mob/abstract/hivemind
+	name = "internal hivemind"
+	see_in_dark = 2
+	see_invisible = SEE_INVISIBLE_LIVING
+
+	var/mob/living/carbon/human/changeling_mob // the head honcho
+	var/list/mob/abstract/hivemind/hivemind = list() // who's in the head honcho's head
+
+/mob/abstract/hivemind/verb/ghost()
+	set category = "OOC"
+	set name = "Ghost"
+	set desc = "Relinquish your life and enter the land of the dead."
+
+	announce_ghost_joinleave(ghostize(1, 0))
+	to_chat(changeling_mob, SPAN_NOTICE("[src] has left our hivemind to join the living dead."))
+	for(var/mob/abstract/hivemind/H in hivemind) // tell the other hiveminds
+		to_chat(H, SPAN_NOTICE("[src] has left our hivemind to join the living dead."))
+
+/mob/abstract/hivemind/proc/add_to_hivemind(var/mob/original_body, var/mob/living/carbon/human/ling)
+	src.name = original_body.real_name
```

Comment:
The `src`s are unnecessary. Although you can keep them if you like them I guess. Here and elsewhere.